internal/provider/provider.go

```diff
@@ -47,15 +42,58 @@ type cacheKey struct {
 	method     string
 }
 
-func (p *provider) login(ctx context.Context, client *api.Client, vaultKubernetesMountPath, roleName, jwt string) (string, error) {
-	p.logger.Debug("vault: performing vault login...")
+func (p *provider) createJWTToken(ctx context.Context, podInfo config.PodInfo) (string, error) {
+	p.logger.Debug("creating service account token bound to pod",
+		"namespace", podInfo.Namespace,
+		"serviceAccountName", podInfo.ServiceAccountName,
+		"podName", podInfo.Name,
+		"podUID", podInfo.UID)
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return "", err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "", err
+	}
+
+	ttl := int64((15 * time.Minute).Seconds())
+	resp, err := clientset.CoreV1().ServiceAccounts(podInfo.Namespace).CreateToken(ctx, podInfo.ServiceAccountName, &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			ExpirationSeconds: &ttl,
+			// TODO: Support audiences as a configurable.
+			//Audiences:         []string{},
```

Comment:
Can you remind me again why we're deferring this? Will this work on EKS without it?

Reply:
I've tested this out on EKS, AKS and GKE now, and it works without setting audiences as long as you set the issuer correctly when configuring vault. Most services default to returning an audience which is the same as the issuer, except EKS which returns `"https://kubernetes.default.svc"`, which I don't quite understand the motivation for.

Legacy style service account tokens (as we were using before) don't have an audience set, so this feature would be purely additive, and I'd like to do a bit more research before implementing and documenting it, but definitely one to follow up on.